packages/frontend/src/components/Game/GameBoard/ResultDisplay/index.jsx

```diff
@@ -1,28 +1,61 @@
 import React from 'react';
 import useGameContext from 'hooks/useGameContext';
+import useFormContext from 'hooks/useFormContext';
+import usePhotosContext from 'hooks/usePhotosContext';
+import Button from 'components/Button';
+import { resetGame, updateStage } from '../../../../contexts/game/actions/';
+import { resetForm } from 'contexts/form/actions';
+import { resetPhotos } from 'contexts/photos/actions';
 
 const ResultDisplay = () => {
   const {
-    state: { winner },
+    state: { winner, stage },
+    dispatch,
   } = useGameContext();
 
+  const formValues = useFormContext();
+  const photoValues = usePhotosContext();
+
+  const handleReset = () => {
+    photoValues.dispatch(resetPhotos());
+    dispatch(resetGame());
+    formValues.dispatch(resetForm());
+  };
+
+  const handlePlayAgain = () => {
+    dispatch(resetGame());
+    dispatch(updateStage(1));
+  };
+
```

Comment:
With multiple contexts, the name of the context could be included in the dispatch variable name so it's more obvious which dispatch is being called.
```
const { dispatch: formDispatch } = useFormContext();
const { dispatch: photosDispatch } = usePhotosContext();

photosDispatch(resetPhotos());
formDispatch(resetForm());
```

Not a huge deal in this case, but I think it makes it more readable in the long run